File: guess.c
#include <stdio.h>
#include <time.h>

int guess(int number_to_guess)
{
    int attempt = 0;
    int x = -1;
    while (scanf("%d", &x)) {
        attempt += 1;
        if (x == number_to_guess)
            return attempt;
        else if (x > number_to_guess)
            printf("Smaller\n");
        else if (x < number_to_guess)
            printf("Bigger\n");
    }
}

int main(int argc, char *argv[])
{
    int number = time(NULL) % 100;
    printf("Guess number I think of\n");
    printf("Your WIN!\nScore:%d",
            guess(number));
    return 0;
}

File: guess.go
package main
import "fmt"
import "math/rand"
import "time"
import "sync"

func main() {
    rand.Seed(time.Now().Unix())
    number := rand.Intn(100)
    score := 0
    c := make(chan int)
    var wg sync.WaitGroup

    go func() {
        for {
            answer := <-c;
            if (answer > number) {
                fmt.Println("Too BIG")
            } else if (answer < number) {
                fmt.Println("Too SMALL")
            } else {
                wg.Done()
            }
        }
    }()
    wg.Add(1)

    go func() {
        fmt.Printf("Guess number I thnik of:")
        for {
            score = score + 1
            var answer int
            fmt.Scanf("%d", &answer)
            c <-answer
        }
    }()

    wg.Wait()
    fmt.Println("You WIN!")
    fmt.Println("Score:", score)
}
